File: panic/golang/services/panic_service.go
package services

import "context"

type PanicService interface {
	CheckPanic(ctx context.Context) string
}

type panicService struct {
}

func NewPanicService() PanicService {
	return &panicService{}
}

func (service *panicService) CheckPanic(ctx context.Context) string {
	panic("test panic")
}

File: panic/rust/src/services/panic_service.rs
pub trait PanicService {
    async fn check_panic(&self) -> String;
}

pub struct PanicServiceImpl {

}

impl PanicServiceImpl {
    pub fn new() -> PanicServiceImpl {
        PanicServiceImpl {  }
    }
}

impl PanicService for PanicServiceImpl {
    async fn check_panic(&self) -> String {
        panic!("something went wrong")
    }
}